runtime/compiler/aarch64/codegen/CallSnippet.cpp

```diff
@@ -0,0 +1,590 @@
+            if (!linkage.getRightToLeft())
+               offset -= 4;
+            if (floatArgNum < linkage.getNumFloatArgRegs())
+               {
+               buffer = storeArgumentItem(TR::InstOpCode::vstrimms, buffer, machine->getRealRegister(linkage.getFloatArgumentRegister(floatArgNum)), offset, cg);
+               }
+            floatArgNum++;
+            if (linkage.getRightToLeft())
+               offset += 4;
+            break;
+         case TR::Double:
+            if (!linkage.getRightToLeft())
+               offset -= 8;
+            if (floatArgNum < linkage.getNumFloatArgRegs())
+               {
+               buffer = storeArgumentItem(TR::InstOpCode::vstrimmd, buffer, machine->getRealRegister(linkage.getFloatArgumentRegister(floatArgNum)), offset, cg);
+               }
+            floatArgNum++;
+            if (linkage.getRightToLeft())
+               offset += 8;
+            break;
+         }
+      }
+
+   return buffer;
+   }
+
+static int32_t instructionCountForArguments(TR::Node *callNode, TR::CodeGenerator *cg)
+   {
+   uint32_t intArgNum=0, floatArgNum=0, count=0;
+   const TR::ARM64LinkageProperties &linkage = cg->getLinkage(callNode->getSymbol()->castToMethodSymbol()->getLinkageConvention())->getProperties();
+   int32_t argStart = callNode->getFirstArgumentIndex();
+
+   for (int32_t i = argStart; i < callNode->getNumChildren(); i++)
+      {
+      TR::Node *child = callNode->getChild(i);
+      switch (child->getDataType())
+         {
+         case TR::Int8:
+         case TR::Int16:
+         case TR::Int32:
+         case TR::Int64:
+         case TR::Address:
+            if (intArgNum < linkage.getNumIntArgRegs())
+               {
+               count++;
+               }
+            intArgNum++;
+            break;
+         case TR::Float:
+         case TR::Double:
+            if (floatArgNum < linkage.getNumFloatArgRegs())
+               {
+               count++;
+               }
+            floatArgNum++;
+            break;
+         }
+      }
+
+   return count;
+   }
+
+static uint32_t encodeHelperBranchAndLink(TR::SymbolReference *symRef, uint8_t *cursor, TR::Node *node, TR::CodeGenerator *cg)
+   {
+   intptrj_t target = (intptrj_t)symRef->getMethodAddress();
+
+   if (cg->directCallRequiresTrampoline(target, (intptrj_t)cursor))
+      {
+      target = TR::CodeCacheManager::instance()->findHelperTrampoline(symRef->getReferenceNumber(), (void *)cursor);
+
+      TR_ASSERT_FATAL(TR::Compiler->target.cpu.isTargetWithinUnconditionalBranchImmediateRange(target, (intptrj_t)cursor),
+                      "Target address is out of range");
+      }
+
+   cg->addExternalRelocation(new (cg->trHeapMemory()) TR::ExternalRelocation(
+                             cursor,
+                             (uint8_t *)symRef,
+                             TR_HelperAddress, cg),
+                             __FILE__, __LINE__, node);
+
+   intptr_t distance = target - (intptr_t)cursor;
```

Comment:
You should make this unsigned.  A bitwise operator (on the next line) on a signed value is implementation dependent.

Reply:
Changed the type to `uintptr_t`.